### PR TITLE
plot_config: fill area between traces

### PR DESCRIPTION
**Pull request by MrKrautee**
## Summary
Improvement and one new feature for plotting indicators.

## What's new?
* not really a fixed bug but a improvement: if plot_config['subplots'] in strategy was given and also subplots on the cmd line (plot-dataframe --indicators2), the plot_config from the strategy was overwritten. the same happend with main_plots. This is fixed now. You can now specify traces in plot_config an cmd line without overwriting them. 
* feature:  Now you can specify two traces to fill the area in between, in plot_config. ie for IchimokuCloud. See the changes in the documentation.

### CODE REVIEW

File: docs/plotting.md [hunk 1]
```diff
@@ -168,6 +168,7 @@ Additional features when using plot_config include:
 
 * Specify colors per indicator
 * Specify additional subplots
+* Specify idicator pairs to fill area in between
```
Comment (xmatthias) [inline docs/plotting.md:171]:
```suggestion
* Specify indicator pairs to fill area in between 
```

File: docs/plotting.md [hunk 2]
```diff
@@ -194,7 +195,13 @@ Sample configuration with inline comments explaining the process:
             "RSI": {
                 'rsi': {'color': 'red'},
             }
-        }
+        },
+	'fill_area': {
+	    "Ichimoku Cloud": { 
+		'traces': ('senkou_a', 'senkou_b'),
+		'color': 'rgba(0,176,246,0.2)', 
+	     },
```
Comment (xmatthias) [inline docs/plotting.md:203]: Are `senkou_a` and `senkou_b` necessary in main_plot as well?

or are they implicitly added to the main plot?
Neither is a problem, but it should be documented...

**Comment by xmatthias** [comment]
>  if plot_config['subplots'] in strategy was given and also subplots on the cmd line (plot-dataframe --indicators2), the plot_config from the strategy was overwritten. the same happend with main_plots. This is fixed now. You can now specify traces in plot_config an cmd line without overwriting them.

Now to be honest, this is intentional - the assumption is that the cli-option takes precedence - as this is the command the user inserted interactively in this moment (-> the user expresses his/her explicit desire to have this configuration in this run).  

Having`--indicators2` do nothing when plot_config is present in the strategy would be highly surprising (and problematic if you wanna have a quick look at an additional indicator).

**Comment by MrKrautee** [comment]
> Having`--indicators2` do nothing when plot_config is present in the strategy would be highly surprising (and problematic if you wanna have a quick look at an additional indicator) - and would completely ignore the user's latest expressed command. 

Hm i think you misunderstand what i have done.
You can still add indicators per cmd line --indicators2. And all indicators get plotted, those present in plot_config and given by --indicators2.

The problem in the recent release is/was that if you use --indicators2, the indicators present in plot_config.subplot gets overwritten, so only indicators given by --indicators2 where plotted and not those from plot_config.subplots.
Here is the problem in the recent release and the cause why plot_config.subplots form the strategy gets dismissed. Better use plot_config['subplots'] .update to add the 'Other' section. 
`        if indicators2:
            plot_config['subplots'] = {'Other': {ind: {} for ind in indicators2}}
`

**Comment by xmatthias** [comment]
i did understand this - but by only adding --indicators2 to subplots (same for main_plot obviously) - you're forcing the user to always take at least the indicators configured in the `plot_config` configuration without a way to remove them.

Assume the plot_config says that the "others" subplot should have `macd` and `macd_signal`.

Now the user calls `plot-dataframe` with `--indicators2 rsi` - currently, "Others" would now only have the RSI indicator.
With this change - "Others" would now have `rsi`, `macd` and `macd_signal` - basically simulating `--indicators2 rsi macd macd_signal`.


since the 2 different indicators (RSI / MACD) don't go well together (different ranges unless the asset is priced between 0 and 100) - you're making one of the 2 types completely unusable.


I think this creates (at least for subplots) more problems and confusion than how it was before (where you had to give the full list of what had to be plotted).

For main plot, i guess it's slightly different - there it can make sense to "quickly add in another indicator" without specifying the full list - mainly because the range is limited by the candles (so the assets price range) anyway.

**Review by xmatthias — COMMENTED** [review]
i know i left this hanging here for quite some time ... sorry for that and for not giving a reason earlier.

The main reason is not the change in behaviour for the overriding - but [freqUI](https://github.com/freqtrade/frequi/). 
echarts currently does not support area-charts natively (
https://github.com/freqtrade/frequi/issues/107 ) (although there are some workarounds) - and since frequi uses the strategy plot configuration (loaded via api), I didn't want to block that development by deciding on a format for `plot-dataframe` which isn't compatible with what's needed for the UI, which would lead to have one type of format for freqUI, and another type of format for plot-dataframe - but the divergence would only be for area plotting, the rest would be aligned - which would lead to very odd user experience. 

I'll try to have a look at this in the comming days (or over the xmas period...)

File: freqtrade/plot/plotting.py [hunk 3]
```diff
@@ -356,6 +364,35 @@ def generate_candlestick_graph(pair: str, data: pd.DataFrame, trades: pd.DataFra
...
+            trace_a = go.Scatter(
+                x=data.date,
+                y=data.get(traces[1]),
+                name=label,
+                fill="tonexty",
+                fillcolor=color,
+                line={'color': 'rgba(255,255,255,0)'},
+            )
+            fig.add_trace(trace_b)
+            fig.add_trace(trace_a)
```
[the remark this answers is not in the document]
Comment (MrKrautee) [reply freqtrade/plot/plotting.py]: this version is outdated. sorry for pushing each change seperately oO :D.

**Comment by MrKrautee** [comment]
I implemented your suggestions.
but tests are missing right now!
I am waiting for your response.

File: freqtrade/plot/plotting.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (xmatthias) [reply freqtrade/plot/plotting.py]: true enough - it's a comment that i wrote a few weeks ago but never finished the review ...

**Pull request closed without merging**